src/dev/platform/f4xx/Timerf4xx.cpp

```diff
@@ -0,0 +1,252 @@
+    } else if (peripheral == TIM9) {
+        interruptIdx = static_cast<uint8_t>(timerInterruptIndex::TIM9_IDX);
+    } else if (peripheral == TIM10) {
+        interruptIdx = static_cast<uint8_t>(timerInterruptIndex::TIM10_IDX);
+    } else if (peripheral == TIM11) {
+        interruptIdx = static_cast<uint8_t>(timerInterruptIndex::TIM11_IDX);
+    } else if (peripheral == TIM12) {
+        interruptIdx = static_cast<uint8_t>(timerInterruptIndex::TIM12_IDX);
+    } else if (peripheral == TIM13) {
+        interruptIdx = static_cast<uint8_t>(timerInterruptIndex::TIM13_IDX);
+    } else if (peripheral == TIM14) {
+        interruptIdx = static_cast<uint8_t>(timerInterruptIndex::TIM14_IDX);
+    } else {
+        interruptIdx = -1;
+    }
+
+    return interruptIdx;
+}
+
+namespace EVT::core::DEV {
+
+Timerf4xx::Timerf4xx(TIM_TypeDef* timerPeripheral, uint32_t clockPeriod) {
+    initTimer(timerPeripheral, clockPeriod);
+    this->halTimer = &halTimers[getTimerInterruptIndex(timerPeripheral)];
+}
+
+void Timerf4xx::initTimer(TIM_TypeDef* timerPeripheral, uint32_t clockPeriod) {
+    this->clockPeriod = clockPeriod;
+    auto& htim = halTimers[getTimerInterruptIndex(timerPeripheral)];
+
+    htim.Instance = timerPeripheral;
+    uint32_t prescaler = SystemCoreClock / 1000;
+    htim.Init.Prescaler = prescaler;// Sets f_CK_PSC to 1000 Hz
+    // Allows period increments of 1 ms with max of 2^(32) ms.
+    htim.Init.CounterMode = TIM_COUNTERMODE_UP;
+    htim.Init.Period = clockPeriod;
+    htim.Init.ClockDivision = TIM_CLOCKDIVISION_DIV1;
+    htim.Init.AutoReloadPreload = TIM_AUTORELOAD_PRELOAD_ENABLE;
+    HAL_TIM_Base_Init(&htim);
+
+    TIM_ClockConfigTypeDef clockConfig = {0};
+    TIM_MasterConfigTypeDef masterConfig = {0};
+
+    clockConfig.ClockSource = TIM_CLOCKSOURCE_INTERNAL;
+    HAL_TIM_ConfigClockSource(&htim, &clockConfig);
+
+    // Timers 9-14 are NOT master mode compatible, so waste of time to go through config
+    if (getTimerInterruptIndex(timerPeripheral) < static_cast<uint8_t>(timerInterruptIndex::TIM9_IDX)) {
+        masterConfig.MasterOutputTrigger = TIM_TRGO_RESET;
+        masterConfig.MasterSlaveMode = TIM_MASTERSLAVEMODE_DISABLE;
+        HAL_TIMEx_MasterConfigSynchronization(&htim, &masterConfig);
+    }
+}
+
+void Timerf4xx::startTimer(void (*irqHandler)(void* htim)) {
+    TIM_TypeDef* timerPeripheral = this->halTimer->Instance;
+    stopTimer();
+    timerInterruptHandlers[getTimerInterruptIndex(timerPeripheral)] = irqHandler;
+    startTimer();
+}
+
+void Timerf4xx::stopTimer() {
+    HAL_TIM_Base_Stop_IT(this->halTimer);
+}
+
+void Timerf4xx::startTimer() {
+    stopTimer();// Stop timer in case it was already running
+
+    auto htim = this->halTimer;
+    // Clear the interrupt flag so interrupt doesn't trigger immediately
+    __HAL_TIM_CLEAR_IT(htim, TIM_IT_UPDATE);
+    HAL_TIM_Base_Start_IT(htim);
+}
+
+void Timerf4xx::reloadTimer() {
```

Comment:
In reloadTimer(), clearing the counter register with this->halTimer->Instance->CNT &= ~(0xFFFFFFFF) can be simplified to just setting the counter to 0: this->halTimer->Instance->CNT = 0;.